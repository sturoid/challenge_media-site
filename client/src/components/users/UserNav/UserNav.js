import React from 'react';
import PropTypes from 'prop-types';
import { NavLink } from 'react-router-dom';

const UserNav = ({ path }) => (
  <nav>
    <NavLink
      exact
      to={`${path}`}
      activeStyle={{ fontWeight: 'bold', color: 'red' }}
    >
      Projects
    </NavLink>
    <NavLink
      to={`${path}/followers`}
      activeStyle={{ fontWeight: 'bold', color: 'red' }}
    >
      Followers
    </NavLink>
  </nav>
);

UserNav.propTypes = {
  path: PropTypes.string.isRequired
};

export default UserNav;
