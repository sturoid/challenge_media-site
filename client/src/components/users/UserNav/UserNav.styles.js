import { css } from 'emotion';
import { greyLight } from '../../../lib/colors.styles';

export const link = css`
  font-weight: bold;
  color: ${greyLight};
  text-decoration: none;
  padding: 20px;
  display: inline-block;
`;

export default link;
