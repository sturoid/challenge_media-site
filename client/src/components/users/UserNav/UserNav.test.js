import React from 'react';
import { HashRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import UserNav from './UserNav';

describe('Rendering', () => {
  it('renders correctly and equals snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <UserNav path="/" />
        </Router>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
