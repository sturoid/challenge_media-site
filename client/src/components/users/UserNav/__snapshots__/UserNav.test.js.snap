// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering renders correctly and equals snapshot 1`] = `
<nav>
  <a
    aria-current="page"
    className="css-ystgrw active"
    href="#/"
    onClick={[Function]}
    style={
      Object {
        "color": "#4E95B1",
      }
    }
  >
    Projects
  </a>
  <a
    aria-current={null}
    className="css-ystgrw"
    href="#//followers"
    onClick={[Function]}
    style={undefined}
  >
    Followers
  </a>
</nav>
`;
